internal/kafka/internal/services/kafka.go

```diff
@@ -1319,3 +1307,27 @@ func buildResourceRecordChange(recordName string, clusterIngress string, action
 
 	return resourceRecordChange
 }
+
+func (k *kafkaService) AssignBootstrapServerHost(kafkaRequest *dbapi.KafkaRequest) error {
+	truncatedKafkaIdentifier := buildTruncateKafkaIdentifier(kafkaRequest)
+	truncatedKafkaIdentifier, replaceErr := replaceHostSpecialChar(truncatedKafkaIdentifier)
+	if replaceErr != nil {
+		return errors.NewWithCause(errors.ErrorGeneral, replaceErr, "generated host is not valid")
+	}
+
+	clusterDNS, err := k.clusterService.GetClusterDNS(kafkaRequest.ClusterID)
+	if err != nil {
+		return errors.NewWithCause(errors.ErrorGeneral, err, "error retrieving cluster DNS")
+	}
+
+	clusterDNS = strings.Replace(clusterDNS, constants2.DefaultIngressDnsNamePrefix, constants2.ManagedKafkaIngressDnsNamePrefix, 1)
+
+	kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
+
+	if k.kafkaConfig.EnableKafkaExternalCertificate {
+		// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
+		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, k.kafkaConfig.KafkaDomainName)
+	}
```

Comment:
if `k.kafkaConfig.EnableKafkaExternalCertificate` is `true` this assignment is ineffective.

Better to change this to
 ```suggestion
	if k.kafkaConfig.EnableKafkaExternalCertificate {
		// If we enable KafkaTLS, the bootstrapServerHost should use the external domain name rather than the cluster domain
		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, k.kafkaConfig.KafkaDomainName)
	} else {
		kafkaRequest.BootstrapServerHost = fmt.Sprintf("%s.%s", truncatedKafkaIdentifier, clusterDNS)
	}
```

Reply:
Hey @ziccardi 
Thanks, great spot :)